### ISSUE TITLE
Out of RAM error on NFC file launch from file browser

### ISSUE THREAD

**Issue by FalsePhilosopher** [DarkFlippers/unleashed-firmware#778]
### Describe the bug.

Using the file browser to run a NFC file will produce a out of RAM error.

### Reproduction

1. Open file browser
2. Scroll to NFC
3. Scroll to NFC file
4. Run the file
5. Out of RAM error

### Target
75-E
_No response_

**Comment by xMasterX** [comment]
Hi, steps to reproduce doesn't work, we are aware of this issue, its caused by RAM fragmentation which can be caused by anything, like any app, etc.

NFC app itself is too large and in such cases it can't fit when RAM is in this state

One user reported that they had NFC not starting even after reboot EVERY time, I suggested replacing/formatting their microsd and reinstalling firmware, and it somehow worked for them
If thats your case too, try this too

In my case I was able to reproduce this issue only 3 times and it was pretty random, currently we have no ideas how to make NFC app smaller to fit in all RAM conditions

**Comment by AztecCodes** [comment]
@xMasterX 

In my case the Error occurs sometimes when I used the Flipper Zero before and used some apps and did different things. The "Out of RAM Error" doesn't occur every time and also not very often, but it does happen from time to time. When I restart the Flipper and directly go for the NFC App, this error never occurs.
However when this Error doesn't occur, the NFC App takes an unusually long to load. It takes like 1-2 seconds till it opens up. It is not insanely long, but it is unusual and it wasn't like this in the older versions. All the other Apps open up instantly, It takes always that long for the NFC App to load. 

I couldn't find a way to reproduce it every time, but I think the chances that it happens are higher when you did use the device for various things before opening the NFC App. 

My experience with your idea for a fix:
I did try out various things you recommended, however the issue still remained on the current release of Unleashed.
With older versions of Unleashed and the current official Flipper Software I don't experience this issue.

When I understand it correctly this issue happens since this release with the updated NFC app. Is there the possibility to add like an option to turn off the additional features if you don't use them? And if you want to use them you can (With experiencing this error from time to time and the slightly longer loading) till a fix for the issue is found.

**Comment by FalsePhilosopher** [comment]
I was getting furi crash errors on boot almost every boot from 73-75 and at random times before the out of ram errors. I formatted the sd card and reinstalled 75 from the SD update function. The furi crash errors at boot have gone away and I can launch a NFC file from the file browser, but like AztecCodes has stated that only work from a fresh boot. If you try to open a few ext apps then launch a nfc file from the file browser you will get a out of ram error. I have now gotten a busfault error instead of a out of ram error as well.
Here is a tar with a .mkv video(My OBS settings were stock and this is what it spit out and github doesn't support uploading mkv files) showing the error. 
[2024-07-13 18-33-32.tar.gz](https://github.com/user-attachments/files/16212331/2024-07-13.18-33-32.tar.gz)

**Comment by AztecCodes** [comment]
I can confirm that I get the busfault error too, but it only happened very few times.
It just happened right now, I didn't use the flipper for like 12 hours and it was on standby, now I tried to start the NFC App from the list and it directly rebooted and showed "busfault" as the error cause.

**Comment by Danucosukosuko** [comment]
Yes, the same. I use apps before NFC, and the same error: Out of memory

**Comment by Kf637** [comment]
I also have the same error.

```
255467 [I][Loader] Loading /ext/apps/NFC/nfc.fap
255625 [I][Elf] Total size of loaded sections: 76318
255628 [I][Loader] Loaded in 161ms
255642 [W][ViewPort] ViewPort lockup: see applications/services/gui/view_port.c:189
255677 [E][Elf] Not enough memory to load section data
255681 [E][Elf] Error loading section '.rodata'
255685 [E][Elf] Not enough memory
```

**Comment by FalsePhilosopher** [comment]
Hold your back arrow for the quick power off function, does it display the crying dolphin or just turn off power? Yesterday I upgraded from 75e to 78e and the issue persisted. I noticed the different power off menu and went about fixing it. Rebooted into DFU mode, repaired to 1.0.1 from qflipper, once it booted after install I formatted the sd card. I then installed 78e from qflipper and I got my crying dolphin back an it doesn't crash launching files from explorer. I have not done extensive testing, just launched a file from browser and micro python which crashed with a bus error the first time but launched the second time.
Edit: After a crash using the change name function and rebooting I don't have the crying dolphin anymore, maybe it's just because I switched from bt screen sharing to an android qflipper instance to usb linux qflipper it's a different screen idk. My screen is broken, so I need to use some form of display forwarding to use my fz.

**Comment by Josue616** [comment]
That doesn't happen with xtreme firmware, but it was archived last July, so there will be no more updates for xtreme firmware, we will only talk about RAM error in NFC app, because xtreme firmware loads the .nfc after opening the list in app, not when you open the application directly.

**Comment by AztecCodes** [comment]
@Josue616 You are right. It started to happen from one specific firmware version of Unleashed and Momentum on. I might be able to find out which one it was. Like the loading time was not as long it is now when starting the NFC app and it didn't crash.

**Comment by Arsi94** [comment]
It seems I've found a solution to this issue. If there are too many NFC files in the folder where they are stored, the Flipper Zero goes into an error state. I moved some of the files I needed into a separate folder, and they launched without any problems. The solution is simply to distribute the NFC files across different folders in smaller amounts. I hope this helps someone!

**Comment by Arsi94** [comment]
Here’s a PowerShell script to make this process faster. Thanks, DeepSeek.

# Specify the path to the root folder
$rootFolder = "C:\Users\DmitryPC\Desktop\FlipperAmiibo-main\Animal_Crossing\Amiibo_Cards\Amiibo_Festival\AC_AF1_Stitches"

# Get all files in all subfolders
Get-ChildItem -Path $rootFolder -Recurse -File | ForEach-Object {
    # Skip files that are already in folders with their names
    if ($_.Directory.Name -ne $_.BaseName) {
        # Create a path for the new folder (in the same directory where the file is located)
        $newFolder = Join-Path -Path $_.DirectoryName -ChildPath $_.BaseName
        
        # Create the folder if it doesn't already exist
        if (-not (Test-Path -Path $newFolder)) {
            New-Item -Path $newFolder -ItemType Directory -Force
        }
        
        # Move the file to the new folder
        Move-Item -Path $_.FullName -Destination $newFolder
    }
}

**Comment by AztecCodes** [comment]
@Arsi94 

Interesting! I did have many NFC files so this could help.
When I get my Flipper I will try it out.

**Comment by mishamyte** [comment]
Looks like the workaround was found. Not sure we can do anything rn, closing it.
